iroha_actor/src/broker.rs

```diff
@@ -88,67 +172,46 @@ impl Broker {
         Self(Arc::new(DashMap::new()))
     }
 
-    fn message_entry(&'_ self, id: TypeId) -> Entry<'_, TypeId, Vec<(TypeId, BrokerRecipient)>> {
-        self.0.entry(id)
-    }
-
     /// Send message via broker
-    pub async fn issue_send<M: BrokerMessage + Send + Sync>(&self, m: M) {
-        let entry = if let Entry::Occupied(entry) = self.message_entry(TypeId::of::<M>()) {
-            entry
-        } else {
-            return;
-        };
-        let send = entry.get().iter().filter_map(|(_, recipient)| {
-            recipient
-                .downcast_ref::<Recipient<M>>()
-                .map(|recipient| recipient.send(m.clone()))
-        });
-        future::join_all(send).await;
-    }
-
-    fn subscribe_recipient<M: BrokerMessage>(&self, recipient: Recipient<M>) {
-        let mut entry = self
-            .message_entry(TypeId::of::<M>())
-            .or_insert_with(|| Vec::with_capacity(1));
-        if entry
-            .iter()
-            .any(|(actor_id, _)| *actor_id == TypeId::of::<Self>())
-        {
-            return;
+    pub async fn issue_send<M: BrokerMessage + Send + Sync>(&self, message: M) {
+        if let Some(subscribers) = self.0.get(&TypeId::of::<M>()) {
+            subscribers.value().publish_message(message).await;
         }
```

Comment:
I think here we can add an `else` and print some warning. Normally, we will not send messages if we have no recipients, or am I wrong?